code/game/objects/items/explosives/plastique.dm

```diff
@@ -76,22 +76,67 @@
 
 		log_explosion("[key_name(user)] planted [src] at [AREACOORD(user.loc)] with [timer] second fuse.")
 
-		target.overlays += image('icons/obj/items/assemblies.dmi', "plastic-explosive_set_armed")
 		user.visible_message("<span class='warning'>[user] plants [name] on [target]!</span>",
 		"<span class='warning'>You plant [name] on [target]! Timer counting down from [timer].</span>")
-		spawn(timer*10)
-			if(plant_target && !plant_target.gc_destroyed)
-				explosion(location, -1, -1, 3)
-				if(istype(plant_target,/turf/closed/wall))
-					var/turf/closed/wall/W = plant_target
-					W.ChangeTurf(/turf/open/floor/plating)
-				else if(istype(plant_target,/obj/machinery/door))
-					qdel(plant_target)
-				else
-					plant_target.ex_act(1)
-				if(plant_target && !plant_target.gc_destroyed)
-					plant_target.overlays -= image('icons/obj/items/assemblies.dmi', "plastic-explosive_set_armed")
-			qdel(src)
+
+		plant_target = target
+		if(ismovableatom(plant_target))
+			var/atom/movable/T = plant_target
+			T.vis_contents += src
+		detonation_pending = addtimer(CALLBACK(src, .proc/detonate), timer*10, TIMER_STOPPABLE)
 
 /obj/item/explosive/plastique/attack(mob/M as mob, mob/user as mob, def_zone)
 	return
+
+/obj/item/explosive/plastique/attack_hand(mob/living/user)
+	if(armed)
+		to_chat(user, "<font color='warning'>Disarm [src] first to remove it!</font>")
+		return
+	return ..()
+
+/obj/item/explosive/plastique/attackby(obj/item/I, mob/user, params)
+	if(ismultitool(I) && armed)
```

Comment:
you could preserve parent logic here if you made them W_CLASS_ABSTRACT or such on being planted to prevent interaction with storages (though this is optional, just a possible improvement)

Reply:
I found ITEM_ABSTRACT in the possible item fields, is that the one?